core/src/main/scala/com/softwaremill/events/model.scala

```diff
@@ -100,7 +100,20 @@ case class PartialEventWithId[U, T](id: Long, eventType: String, aggregateType:
 }
 
 case class StoredEvent(id: Long, eventType: String, aggregateType: String, aggregateId: Long, aggregateIsNew: Boolean,
-  created: OffsetDateTime, userId: Long, txId: Long, eventJson: String)
+  created: OffsetDateTime, userId: Long, txId: Long, eventJson: String){
+
+  def toEvent(clsOpt: Option[Class[_]]): Event[Any] = {
+    //TODO: capture formats with event, maybe in registry
+
+    clsOpt match {
+      case Some(clazz) =>
```

Comment:
definitely not obvious ;) you are getting a compiler warning here - and for a good reason (should be an error in fact ;) ). Anyway, it's certainly possible that there's no model update for an event - in which case we should just skip it. Either handle the option higher (in whatever calls `toEvent`), or return an `Option[Event]`.